contracts/src/Zappers/Modules/FlashLoans/BalancerFlashLoan.sol

```diff
@@ -34,19 +29,22 @@ contract BalancerFlashLoan is IFlashLoanRecipient, IFlashLoanProvider {
         if (_operation == Operation.OpenTrove) {
             ILeverageZapper.OpenLeveragedTroveParams memory openTroveParams =
                 abi.decode(_params, (ILeverageZapper.OpenLeveragedTroveParams));
-            userData = abi.encode(_caller, _operation, openTroveParams);
+            userData = abi.encode(_operation, openTroveParams);
         } else if (_operation == Operation.LeverUpTrove) {
             ILeverageZapper.LeverUpTroveParams memory leverUpTroveParams =
                 abi.decode(_params, (ILeverageZapper.LeverUpTroveParams));
-            userData = abi.encode(_caller, _operation, leverUpTroveParams);
+            userData = abi.encode(_operation, leverUpTroveParams);
         } else if (_operation == Operation.LeverDownTrove) {
             ILeverageZapper.LeverDownTroveParams memory leverDownTroveParams =
                 abi.decode(_params, (ILeverageZapper.LeverDownTroveParams));
-            userData = abi.encode(_caller, _operation, leverDownTroveParams);
+            userData = abi.encode(_operation, leverDownTroveParams);
         } else {
             revert("LZ: Wrong Operation");
         }
 
+        // This will be used by the callback below no
+        receiver = IFlashLoanReceiver(msg.sender);
+
         vault.flashLoan(this, tokens, amounts, userData);
```

Comment:
Personally, I would have put the reset of `receiver` here, after the return of `vault.flashLoan()`. IMO it's more readable when the assignments are kept in the same function (easy to pair them together).

Plus, the current implementation relies on Balancer not returning successfully without having called us back. Now, in this case, it's easy to see by reading the Balancer source code that that's impossible, but in general, I prefer to minimize relying on external "promises" as much as possible.